.evergreen/config.yml

```diff
@@ -54,58 +62,105 @@ functions:
         add_expansions_to_env: true
         working_dir: "src/${DIR}/${REPO_NAME}"
         binary: bash
-        env:
-          atlas: ${workdir}/src/atlas/bin/atlas
         args:
           - ../run.sh
 
-  "setup atlas cli":
+  "setup local atlas":
     - command: subprocess.exec
       type: setup
       retry_on_failure: true
       params:
         add_expansions_to_env: true
         working_dir: "src"
         binary: bash
-        env:
-          atlas: ${workdir}/src/atlas/bin/atlas
         args:
           - .evergreen/provision-atlas.sh
 
+  "setup remote atlas":
+    - command: subprocess.exec
+      type: setup
+      params:
+        add_expansions_to_env: true
+        working_dir: "src"
+        binary: bash
+        args: [.evergreen/setup-remote.sh]
+
 pre_error_fails_task: true
 pre:
   - func: "fetch source"
-  - func: "setup atlas cli"
+  - func: "fetch secrets"
 
 tasks:
-  - name: test-semantic-kernel-python
+  - name: test-semantic-kernel-python-local
+    commands:
+      - func: "fetch repo"
+      - func: "setup local atlas"
+      - func: "execute tests"
+
+  - name: test-semantic-kernel-python-remote
+    commands:
+      - func: "fetch repo"
+      - func: "setup remote atlas"
+      - func: "execute tests"
+
+  - name: test-semantic-kernel-csharp-local
+    commands:
+      - func: "fetch repo"
+      - func: "setup local atlas"
+      - func: "execute tests"
+
+  - name: test-semantic-kernel-csharp-remote
+    commands:
+      - func: "fetch repo"
+      - func: "setup remote atlas"
+      - func: "execute tests"
+
+  - name: test-langchain-python-local
     commands:
       - func: "fetch repo"
+      - func: "setup local atlas"
       - func: "execute tests"
 
-  - name: test-semantic-kernel-csharp
+  - name: test-langchain-python-remote
```

Comment:
Do we need to add tags or can we do regex matching in evergreen to make sure these aren't run on pull requests